marketplace-service/src/main/java/ch/akros/marketplace/service/controller/TopicController.java

```diff
@@ -3,53 +3,58 @@
 import ch.akros.marketplace.api.TopicsApi;
 import ch.akros.marketplace.api.model.*;
 import ch.akros.marketplace.service.service.TopicService;
+
 import lombok.extern.slf4j.Slf4j;
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.http.HttpStatus;
 import org.springframework.http.ResponseEntity;
-import org.springframework.web.bind.annotation.RestController;
+import org.springframework.web.bind.annotation.*;
+import org.springframework.web.multipart.MultipartFile;
 
 @RestController
 @Slf4j
 public class TopicController implements TopicsApi {
-  @Autowired
-  private TopicService topicService;
+    @Autowired
+    private TopicService topicService;
 
-  @Override
-  public ResponseEntity<TopicLoadResponseDTO> topicsTopicIdGet(Long topicId) {
-    try {
-      log.debug("TopicController.TopicTopicIdGet() called");
-      TopicLoadResponseDTO topic = topicService.loadTopic(topicId);
-      return ResponseEntity.status(HttpStatus.OK).body(topic);
-    } catch (Exception ex) {
-      log.error(ex.getMessage(), ex);
-      return ResponseEntity.status(HttpStatus.INTERNAL_SERVER_ERROR).build();
+    @Override
+    public ResponseEntity<TopicLoadResponseDTO> topicsTopicIdGet(Long topicId) {
+        try {
+            log.debug("TopicController.TopicTopicIdGet() called");
+            TopicLoadResponseDTO topic = topicService.loadTopic(topicId);
+            return ResponseEntity.status(HttpStatus.OK).body(topic);
+        } catch (Exception ex) {
+            log.error(ex.getMessage(), ex);
+            return ResponseEntity.status(HttpStatus.INTERNAL_SERVER_ERROR).build();
+        }
     }
-  }
 
-  @Override
-  public ResponseEntity<Void> topicsPost(TopicSaveRequestDTO topicSaveRequestDTO) {
-    try {
-      log.debug("TopicController.topicsPost() called");
-      topicService.saveTopic(topicSaveRequestDTO);
-      return ResponseEntity.status(HttpStatus.OK).build();
-    } catch (Exception ex) {
-      log.error(ex.getMessage(), ex);
-      return ResponseEntity.status(HttpStatus.INTERNAL_SERVER_ERROR).build();
+    @PostMapping(value = "/topics", consumes = { "multipart/form-data" })
+    public ResponseEntity<Void> createTopic(@RequestPart("topics") String topicSaveRequestDTO,
+                                            @RequestPart(value = "thumbnail", required = false) MultipartFile thumbnail,
+                                            @RequestPart(value = "images", required = false) MultipartFile[] images) {
+        try {
+            log.debug("TopicController.topicsPost() called");
+            topicService.saveTopic(topicSaveRequestDTO, images, thumbnail);

```

Comment:
Before saving the images, would it be possible to verify that the files uploaded are indeed images? Or maybe create an issue to deal with that later.